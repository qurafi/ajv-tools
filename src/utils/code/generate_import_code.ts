const json = JSON.stringify;

export function generateDynamicImportsCode(
	refs: string[],
	map = (ref: string) => ref,
	export_name?: string,
) {
	const named_import = export_name ? `.then(v=>v[${json(export_name)}])` : "";
	const refsKeys = refs.map((ref) => {
		return `${json(ref)}:() => import(${json(map(ref))})${named_import}`;
	});
	return `export default {${refsKeys.join(",")}}`;
}
