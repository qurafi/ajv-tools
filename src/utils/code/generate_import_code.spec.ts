import { expect, test } from "vitest";
import { generateDynamicImportsCode } from "./generate_import_code.js";

const test_refs = ["test", "test2"];

test("generate code for list of dynamic import with mapping", async () => {
    const code = generateDynamicImportsCode(test_refs, (v) => `someprefix/${v}`);
    expect(code).toMatchSnapshot();
});
