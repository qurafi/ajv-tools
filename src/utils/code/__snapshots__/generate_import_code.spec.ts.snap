// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generate code for list of dynamic import with mapping 1`] = `"export default {\\"test\\":() => import(\\"someprefix/test\\"),\\"test2\\":() => import(\\"someprefix/test2\\")}"`;
